src/apify/_actor.py

```diff
@@ -270,8 +270,8 @@ async def finalize() -> None:
             self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in IPython')
         elif os.getenv('PYTEST_CURRENT_TEST', default=False):  # noqa: PLW1508
             self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in an unit test')
-        elif hasattr(asyncio, '_nest_patched'):
-            self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in a nested event loop')
+        elif os.getenv('SCRAPY_SETTINGS_MODULE'):
```

Comment:
Please mention that we now depend on this to be present in the upgrading guide, just to be sure.

Reply:
I am not sure about the upgrading guide mentioning minor versions. Is it okay if I add it to the Scrapy guide doc page?